members/forms.py

```diff
@@ -1,19 +1,36 @@
 from django import forms
+from django.utils import timezone
 from django.contrib.auth.forms import UserCreationForm
 from django.contrib.auth.models import User
+from iquise.utils import random_password
 
-from members.models import Person
+def this_year():
+    return timezone.now().year
 
+class JoinForm(forms.Form):
+    LEVELS = ("Highschool", "Undergraduate", "Graduate", "PostDoc", "Professor", "Professional", "Retired")
+    first_name = forms.CharField(max_length=30)
+    last_name = forms.CharField(max_length=30)
+    email = forms.EmailField(help_text='Please use your university email if possible.')
+    affiliation = forms.CharField(max_length=30)
+    graduation_year = forms.IntegerField(initial=this_year, min_value=1900, help_text='Past, future or expected.')
+    level = forms.ChoiceField(initial=LEVELS[1], choices=zip(LEVELS, LEVELS))
 
-class PersonForm(forms.ModelForm):
-    def __init__(self, *args, **kwargs):
-        super(PersonForm, self).__init__(*args, **kwargs)
-        self.fields['email'].required = True
-        self.fields['lab'].required = True
-        self.fields['lab'].label = 'Affiliation/Lab'
-    class Meta:
-        model=Person
-        fields = ('first_name','last_name','email','lab','year')
+    def clean_email(self):
+        return self.cleaned_data["email"].lower()
+
+    def save(self):
+        u = User.objects.create(
+            username=self.cleaned_data["email"],
+            email=self.cleaned_data["email"],
+            first_name=self.cleaned_data["first_name"],
+            last_name=self.cleaned_data["last_name"],
+            password=random_password(),
+        )
+        u.profile.affiliation = self.cleaned_data["affiliation"]
+        u.profile.graduation_year = self.cleaned_data["graduation_year"]
+        u.profile.level = self.cleaned_data["level"]
+        u.profile.save()
```

Comment:
maybe a future todo but it'd be nice to eventually send a confirmation email to them with a copy of what they submitted... to catch things like typos (if they make an email typo they won't get a confirmation email)

Reply:
I generally dislike when sites do that without a better reason. If anything we could just embrace people having accounts and passwords now. Then send an email to verify email (which is already better security to prove they own that email)